presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -331,4 +339,32 @@ private Estimate summarizePartitionStatistics(
                             .collect(toImmutableSet()));
         }
     }
+
+    @VisibleForTesting
+    static List<HivePartition> getPartitionsSample(List<HivePartition> partitions, int sampleSize)
+    {
+        checkArgument(sampleSize > 0, "sampleSize is expected to be equal to or greater than zero");
+
+        if (partitions.size() <= sampleSize) {
+            return partitions;
+        }
+
+        Map<String, Integer> hashes = computeHashes(partitions);
+        List<HivePartition> partitionsSortedByHash = new ArrayList<>(partitions);
+        Comparator<HivePartition> comparator = Comparator
+                .comparingInt((HivePartition partition) -> hashes.get(partition.getPartitionId()))
+                .thenComparing(HivePartition::getPartitionId);
+        partitionsSortedByHash.sort(comparator);
```

Comment:
I think going through the hash map is unnecessary. What about shortening the code all this code to:
```
HashFunction hashFunction = goodFastHash(32);
return partitions.stream()
        .map(partition -> immutableEntry(partition, hashFunction.hashUnencodedChars(partition.getPartitionId()).asInt()))
        .sorted(Comparator.<Entry<HivePartition, Integer>, Integer>comparing(Entry::getValue)
                .thenComparing(entry -> entry.getKey().getPartitionId()))
        .limit(sampleSize)
        .map(Entry::getKey)
        .collect(toImmutableList());
```

Reply:
Actually yes, i'm creating a copy anyway =)

Reply:
Yes, both approaches copy the data.

Reply:
I though that creating 10_000 objects will be more expensive than managing the hashmap. But bechmark shows completely oposite. It is 1.7ms for the `immutableEntry`  vs 6ms for the `HashMap`